### ISSUE TITLE
hide dialogue overlay box "you have reached the end of your selection"

### ISSUE THREAD

**Issue by naisanzaa** [darktable-org/darktable#17763]
**Is your feature request related to a problem? Please describe.**

<img width="341" alt="image" src="https://github.com/user-attachments/assets/1405bc3c-aa3d-4673-be99-05d744d10b91">

**Describe the solution you'd like**

It would be nice to hide these dialogue boxes because they are big and feel more like they get in the way then being helpful to the user.

**Alternatives**

**Additional context**

**Comment by ralfbrown** [comment]
The code should probably remember whether the message has been issued since the image became active so that it only shows once even with repeated keypresses.  I'd say its OK to see the "reached start/end" message again if the user navigated to some other image and then back to the first/last in the selection.

**Comment by naisanzaa** [comment]
How do I hide those control log messages all together? 

Or, setting a timeout for these log messages low enough so it doesn't show at all. 

Or, an option to offset where the log message shows on the screen. So I can offset them so far away I don't see it in the app at all.

**Comment by naisanzaa** [comment]
> @naisanzaa :
> 
> > How do I hide those control log messages all together?
> 
> You can't, but this will be fixed in next release.

fixed as in having the option to turn this logging off?

**Comment by naisanzaa** [comment]
I saw the new update. 

Can we have a maximum of 1 line? (or just an option to completely turn this off)

It still posts many multiple lines to the screen when you hit the beginning of the selection, and the end of the selection.

**Comment by nepella** [comment]
I suppressed the messages by sticking this embarrassing hackiness in my CSS:

```css
.dt_messages {
  color: alpha(white, 0);
  background-color: alpha(white, 0);
}
```

The downside is that it also hides toasts, **&lt;whining&gt;** but it's worth it not to have my culling and full-preview views disfigured with the YoU hAvE rEaChEd ThE EnD Of yOuR SeLeCtiOn message whenever I forget the precise beginning and ending of my selection and arrow one too far. I'm not that sharp, but even I would get the message fine with 200-millisecond `first` and `last` messages. I can't imagine what Darktable user needs **five seconds** with this level of spoon-fed verbosity to understand how navigation works.**&lt;/whining&gt;** ;)

**Comment by naisanzaa** [comment]
> I suppressed the messages by sticking this embarrassing hackiness in my CSS:
> 
> .dt_messages {
>   color: alpha(white, 0);
>   background-color: alpha(white, 0);
> }
> The downside is that it also hides toasts, **<whining>** but it's worth it not to have my culling and full-preview views disfigured with the YoU hAvE rEaChEd ThE EnD Of yOuR SeLeCtiOn message whenever I forget the precise beginning and ending of my selection and arrow one too far. I'm not that sharp, but even I would get the message fine with 200-millisecond `first` and `last` messages. I can't imagine what Darktable user needs **five seconds** with this level of spoon-fed verbosity to understand how navigation works.**</whining>** ;)

yeah the messages would be better located _anywhere_ not on the pictures itself.

**Comment by nepella** [comment]
My previous workaround stopped working, and I very much didn't want to have to rebuild Darktable, so I wrote a shim for the logging function that drops these messages, and a script that uses `LD_PRELOAD` to run Darktable with the shim. I tested this on Linux.

---

`darktable-shim.c`:

```c
#include <dlfcn.h>     /* dlsym(), RTLD_NEXT */
#include <stdarg.h>    /* va_end(), va_list, va_start() */
#include <string.h>    /* strncmp() */

static const char cursed[] = "you have reached";

/* Overriding <$darktable/src/control/control.c>. */
void dt_control_log(const char *msg, ...) {
    static void *(*original_dt_control_log)(const char *, ...) = NULL;
    va_list ap;

    if (!original_dt_control_log) {
        original_dt_control_log = dlsym(RTLD_NEXT, "dt_control_log");
    }

    if (strncmp(msg, cursed, strlen(cursed))) {
        va_start(ap, msg);
        original_dt_control_log(msg, ap);
        va_end(ap);
    }
}

void _init(void) {
}
```

`darktable`:

```sh
#!/bin/bash
set -eu

declare -r EX_SOFTWARE=70

inode() { stat -Lc %d:%i -- "$1"; }

((SHLVL > 50)) && exit $EX_SOFTWARE

# Find a binary that is not this script.
my_id=$(inode "$0")
while IFS= read -r path; do
    if [[ "$my_id" != "$(inode "$path")" ]]; then
        bin=$path
        break
    fi
done < <(type -ap darktable)
[[ -z ${bin-} ]] && exit $EX_SOFTWARE

mydir=$(realpath -- "$(dirname -- "$BASH_SOURCE")/../lib")
LD_PRELOAD="$mydir/darktable-shim.so" exec "$bin" "$@"
```

`Makefile`:

```makefile
PREFIX = ~/.local
CFLAGS = -Wall -fPIC -DPIC

.PHONY: all
all: darktable-shim.so

.PHONY: install
install: all
	mkdir -p $(PREFIX)/bin $(PREFIX)/lib
	install -t $(PREFIX)/bin darktable
	install -m 0644 -t $(PREFIX)/lib darktable-shim.so

.PHONY: clean
clean:
	rm -f darktable-shim.o

.PHONY: distclean
distclean: clean
	rm -f darktable-shim.so

darktable-shim.so: darktable-shim.o
	ld -L/usr/lib -shared -o $@ $< -ldl
```
